apps/backend/src/services/accounts.js

```diff
@@ -176,137 +172,141 @@ async function hydrateDemoData({ userId }) {
+  const user = await UserModel.findById(userId)
 
   // create demo tag
-  const demoTag = new TagModel({
+  const demoTag = await new TagModel({
     name: 'DEMO',
     questions: [],
     user: userId,
-  })
-  await demoTag.save()
+  }).save()
+  user.tags.push(demoTag.id)
 
   // create demo questions with demo tag from demo data above
-  for (let i = 0; i < titles.length; i += 1) {
-    const newQuestion = new QuestionModel({
-      tags: [demoTag.id],
-      title: titles[i],
-      type: types[i],
-      user: userId,
-      versions: [
-        {
-          content: content[i],
-          description: content[i]
-            .replace(/(\${2})[^]*?[^\\]\1/gm, '$FORMULA$')
-            .replaceAll('<br>', '')
-            .match(/[\p{L}\p{N}\p{P}\p{M}\s]|[$Formula$]|[(0-9)+. ]|[- ]/gu)
-            .join(''),
-          options: options[i],
-          files: [],
-          solution: undefined,
-        },
-      ],
+  const demoQuestions = await Promise.all(
+    DEMO_QUESTIONS.map(async ({ title, type, content, options }) => {
+      const newQuestion = await new QuestionModel({
+        tags: [demoTag.id],
+        title,
+        type,
+        user: userId,
+        versions: [
+          {
+            content,
+            description: content
+              .replace(/(\${2})[^]*?[^\\]\1/gm, '$FORMULA$')
+              .replaceAll('<br>', '')
+              .match(/[\p{L}\p{N}\p{P}\p{M}\s]|[$Formula$]|[(0-9)+. ]|[- ]/gu)
+              .join(''),
+            options,
+            files: [],
+            solution: undefined,
+          },
+        ],
+      }).save()
+
+      demoTag.questions.push(newQuestion.id)
+      user.questions.push(newQuestion.id)
+
+      return newQuestion
     })
-    demoQuestions.push(newQuestion)
-    demoTag.questions.push(newQuestion.id)
-    user.questions.push(newQuestion.id)
-    await newQuestion.save()
-  }
-  await user.save()
-  await demoTag.save()
+  )
+
+  await Promise.all([user.save(), demoTag.save()])
 
   // create instances for session population
   const sessionId1 = ObjectId()
-  const sessionId2 = ObjectId()
-  const instanceIds = [] // instance array containing 3 instances of each question
-
-  const questionInstanceCreator = async (question, sessionId) => {
-    const instance = new QuestionInstanceModel({
-      question,
-      session: sessionId,
-      user: userId,
-      version: 0,
-    })
-    await instance.save()
-    question.instances.push(instance.id)
-    return instance.id
-  }
 
-  // instance array containing instances for questions [1, 1, 2, 2, 3, 3, 4, 4, 1, 2, 3, 4]
-  for (let i = 0; i < titles.length; i += 1) {
-    instanceIds.push(await questionInstanceCreator(demoQuestions[i], sessionId1))
-    instanceIds.push(await questionInstanceCreator(demoQuestions[i], sessionId1))
-  }
+  // prepare 8 question instances for the session without results (in pairs of two)
+  const questionInstances = await demoQuestions.reduce(async (acc, question) => {
```

Comment:
I adapted it like this for the two sessions for simplicity. The order is not consistent across users (q4 might be a SC for one, and NR for another). But that actually does not matter, so it was much easier to do everything "in parallel" instead of sequentially with for loops.